sound/soc/sof/ipc.c

```diff
@@ -689,42 +831,62 @@ int snd_sof_ipc_set_get_comp_data(struct snd_sof_ipc *ipc,
 {
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
+	struct sof_ipc_ctrl_data_params sparams;
+	struct sof_ipc_fw_ready *ready = &sdev->fw_ready;
+	struct sof_ipc_fw_version *v = &ready->version;
 	int send;
 	int err = 0;
 
+	send = sof_get_ctrl_ipc_dir(ctrl_type);
```

Comment:
Ok, I see, you use "send" twice now

Reply:
So defining functions for send dir and send params setting was more to make the whole thing readable for others. I'm using the "send" twice and the params setting is more to get rid of huge amount of local variables (this was comment from Pierre). I think the send message function is now pretty clean but you are right that it is difficult to justify the new functions other than making things more readable. If you think it is better just to include everything in this one function, I'm fine with it... @plbossart thoughts?

Reply:
@juimonen I don't really have a recommendation, but I am not a big fan of a function that only does a switch case, and at the same time the get_set way of doing things doesn't have a direct mapping with other protocols such as rpmsg or mailboxes. Need more coffee...

Reply:
@juimonen how about adding this to the signature of snd_sof_ipc_set_get_comp_data(). The caller of the function knows if it is a get or a set and can just set it appropriately

Reply:
@ranj063 ok did that